src/chunked_scatter/chunked_scatter.py

```diff
@@ -19,132 +19,134 @@
-        if fields[0] not in ["browser", "track"] and len(line) >= 3:
-            start = int(fields[1])
-            end = int(fields[2])
-            name = fields[0]
-            position = start
-            # This will cause the last chunk to be between 0.5 and 1.5
-            # times the chunk_size in length, this way we avoid the
-            # possibility that the last chunk ends up being to small
-            # (eg. 1+overlap bases).
-            while position + chunk_size*1.5 < end:
-                if position-overlap <= start:
-                    yield [name, start, int(position+chunk_size)]
-                else:
-                    yield [name, int(position-overlap),
-                           int(position+chunk_size)]
-                position += chunk_size
-            if position-overlap <= start:
-                yield [name, start, end]
+    for contig, start, end in regions:
+        position = start
+        # This will cause the last chunk to be between 0.5 and 1.5
+        # times the chunk_size in length, this way we avoid the
+        # possibility that the last chunk ends up being to small
+        # (eg. 1+overlap bases).
+        while position + chunk_size * 1.5 < end:
+            if position - overlap <= start:
+                yield BedRegion(contig, start, int(position + chunk_size))
             else:
-                yield [name, int(position-overlap), end]
+                yield BedRegion(contig, int(position - overlap),
+                                int(position + chunk_size))
+            position += chunk_size
+        if position - overlap <= start:
+            yield BedRegion(contig, start, end)
+        else:
+            yield BedRegion(contig, int(position - overlap), end)
 
 
-def input_is_bed(filename: Path):
-    """
-    Check whether or not the input file is a bed file.
-    :param filename: The filename.
-    """
-    if filename.suffix == ".bed":
-        return True
-    elif filename.suffix == ".dict":
-        return False
-    else:
-        raise ValueError(
-            "Only files with .bed or .dict extensions are supported.")
+def chunked_scatter(regions: Iterable[BedRegion],
+                    chunk_size: int,
+                    overlap: int,
+                    minimum_base_pairs: int,
+                    split_contigs: bool = False,
+                    ) -> Generator[List[BedRegion], None, None]:
+    current_scatter_size = 0
+    current_contig = None
+    chunk_list: List[BedRegion] = []
+    for chunk in region_chunker(regions, chunk_size, overlap):
+        # If the next chunk is on a different contig
+        if chunk.contig != current_contig or split_contigs:
+            current_contig = chunk.contig
+            # and the current bed file contains enough bases
+            if current_scatter_size >= minimum_base_pairs:
+                yield chunk_list
+                chunk_list = []
+                current_scatter_size = 0
```

Comment:
Yes indeed, I think I should call it `contigs_can_be_split`.

Reply:
> In addition, we can speed this up by like 0.01 seconds per iteration in case split_contigs is true

Yes! More performance pedantry! I will make this change!